test/cases/cast.zig

```diff
@@ -75,3 +74,20 @@ test "string literal to &const []const u8" {
     const x: &const []const u8 = "hello";
     assert(mem.eql(u8, *x, "hello"));
 }
+
+test "implicitly cast from T to %?T" {
+    castToMaybeTypeError();
+    comptime castToMaybeTypeError();
+}
+const A = struct {
+    a: i32,
+};
+fn castToMaybeTypeError() {
```

Comment:
I believe this function is only exercising the compile time aspects of this cast. if you pass in the value 1 as a parameter then this will test both compile time and runtime implicit casts.